sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4063,6 +4063,19 @@ object SQLConf {
       .checkValues(Set("none", "zstd", "lz4"))
       .createWithDefault("none")
 
+  val ARROW_EXECUTION_ZSTD_COMPRESSION_LEVEL =
+    buildConf("spark.sql.execution.arrow.zstd.compressionLevel")
+      .doc("Compression level for Zstandard (zstd) codec when compressing Arrow IPC data. " +
+        "This config is only used when spark.sql.execution.arrow.compressionCodec is set to " +
+        "'zstd'. Valid values are integers from 1 (fastest, lowest compression) to 22 " +
+        "(slowest, highest compression). The default value 3 provides a good balance between " +
+        "compression speed and compression ratio.")
+      .version("4.1.0")
+      .intConf
+      .checkValue(level => level >= 1 && level <= 22,
+        "Zstd compression level must be between 1 and 22")
+      .createWithDefault(3)
```

Comment:
I know the underlying zstd lib uses 3 as the default value. for `spark.io.compression.zstd.level`, it chooses 1 as the default value, for our internal testing, 1 definitely is the best choice in terms of the immediate data exchange case (i.e., shuffle).

I'm not worried about either 1 or 3 as the default value, but it would be great if you could provide some testing reports as a reference.

Reply:
It is picked as the default compression level is 3 in the `ZstdCompressionCodec`. I have no opinion on the default value, though.

Reply:
> as the default compression level is 3 in the `ZstdCompressionCodec.`

But I only found such info in the `ZstdCompressionCodec.`
```
  // Default compression level for zstd compression to 1 because it is
  // fastest of all with reasonably high compression ratio.
  private val level = conf.get(IO_COMPRESSION_ZSTD_LEVEL)
```

Reply:
I mean Arrow `ZstdCompressionCodec`...